src/gcode-machine-control.cc

```diff
@@ -534,6 +551,14 @@ const char *GCodeMachineControl::Impl::special_commands(char letter,
   case 119: mprint_endstop_status(); break;
   case 120: pause_enabled_ = true; break;
   case 121: pause_enabled_ = false; break;
+  case 181: {
+    const char *const after_pair = parser_->ParsePair(remaining, &letter, &value, msg_stream_);
```

Comment:
I suggest to move the whole logic in a separate `handle_set_lookahead()` function.
(see `set_spindle_on()` as an example).